### PR TITLE
Token global distribution

### PR DESCRIPTION
**Pull request by Peartes**
This commit is going to 

- [x] Rename Project Token Global Distribution to Project Token Distribution per Stakeholder Group
- [x] There should be 4 columns (Stakeholder (as plain text input field), Amount, Cliff (in days), Vest(in days))
- [x] Cliff and vest can be 0. 
- [x] Amount can be 0 as well.

A new look is available below

image

image

And the summary page
image

### CODE REVIEW

File: src/newSeed/stage3.ts [hunk 1]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
-          message = "Please enter a value for Category";
-        } else if (!tokenDistrb.amount || tokenDistrb.amount === "0") {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Amount`;
-        } else if (!(tokenDistrb.lockup > 0)) {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Lock-up`;
+      this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: { stakeHolder: string, amount: string, cliff: number, vest: number }) => {
+        if (!tokenDistrb.stakeHolder) {
+          message = "Please enter a value for Stakeholder";
+        } else if (!tokenDistrb.amount) {
+          message = `Please enter a value for Category ${tokenDistrb.stakeHolder} Amount`;
```
Comment (dkent600) [inline src/newSeed/stage3.ts:82]: Instead of "Category ${tokenDistrb.stakeHolder}"  should just be "Stakeholder"

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/newSeed/stage3.ts (hunk 1 above)
Comment (dkent600) [reply src/newSeed/stage3.ts:82]: Hmm, actually, just make it "${tokenDistrb.stakeHolder}"

Comment (dkent600) [reply src/newSeed/stage3.ts:82]: I just worry that stakeHolder can be 30 chars, which is long for such a message....

Comment (Peartes) [reply src/newSeed/stage3.ts:82]: I agree, stakeholder might be long making the message even longer

File: src/newSeed/stage3.ts [hunk 2]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
-      this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: { category: string, amount: string, lockup: number }) => {
-        if (!tokenDistrb.category) {
-          message = "Please enter a value for Category";
-        } else if (!tokenDistrb.amount || tokenDistrb.amount === "0") {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Amount`;
-        } else if (!(tokenDistrb.lockup > 0)) {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Lock-up`;
+      this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: { stakeHolder: string, amount: string, cliff: number, vest: number }) => {
+        if (!tokenDistrb.stakeHolder) {
+          message = "Please enter a value for Stakeholder";
```
Comment (dkent600) [inline src/newSeed/stage3.ts:80]: My final request was to make it  "${tokenDistrb.stakeHolder}".  Thing is, we also have to identify the row it is in, yes?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/newSeed/stage3.ts (hunk 2 above)
Comment (dkent600) [reply src/newSeed/stage3.ts:80]: the lines below say "Category", should also be "${tokenDistrb.stakeHolder}"

Comment (Peartes) [reply src/newSeed/stage3.ts:80]: Should we use the position like StakeHolder 1, Stakeholder 2 based on the position in the array ?

Comment (dkent600) [reply src/newSeed/stage3.ts:80]: What does it look like when you enter 30 chars?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/newSeed/stage3.ts [hunk 3]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
-      this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: { category: string, amount: string, lockup: number }) => {
-        if (!tokenDistrb.category) {
-          message = "Please enter a value for Category";
-        } else if (!tokenDistrb.amount || tokenDistrb.amount === "0") {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Amount`;
-        } else if (!(tokenDistrb.lockup > 0)) {
-          message = `Please enter a number greater than zero for Category ${tokenDistrb.category} Lock-up`;
+      this.seedConfig.tokenDetails.tokenDistrib.forEach((tokenDistrb: { stakeHolder: string, amount: string, cliff: number, vest: number }) => {
+        if (!tokenDistrb.stakeHolder) {
+          message = "Please enter an Amount for Stakeholder";
```
Comment (dkent600) [inline src/newSeed/stage3.ts:80]: "Please enter a name for the Stakeholder"

File: src/newSeed/stage3.ts [hunk 4]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
+        } else if (!tokenDistrb.amount) {
+          message = `Please enter an Amount for ${tokenDistrb.stakeHolder} Amount`;
```
Comment (dkent600) [inline src/newSeed/stage3.ts:82]: `Please enter an amount for ${tokenDistrb.stakeHolder}`

File: src/newSeed/stage3.ts [hunk 5]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
+        } else if (!tokenDistrb.cliff) {
+          message = `Please enter an Amount for ${tokenDistrb.stakeHolder} Cliff`;
```
Comment (dkent600) [inline src/newSeed/stage3.ts:84]: `Please enter the cliff for ${tokenDistrb.stakeHolder}`

File: src/newSeed/stage3.ts [hunk 6]
```diff
@@ -75,13 +75,15 @@ export class Stage3 extends BaseStage {
...
+        } else if (!tokenDistrb.vest) {
+          message = `Please enter an Amount for ${tokenDistrb.stakeHolder} Vest`;
```
Comment (dkent600) [inline src/newSeed/stage3.ts:86]: `Please enter the vesting period for ${tokenDistrb.stakeHolder}`

**Review by dkent600 — CHANGES_REQUESTED** [review]

**Review by dkent600 — APPROVED** [review]

**Pull request merged**